src/WireFormat.h

```diff
@@ -1605,6 +1648,182 @@ struct TakeIndexletOwnership {
+        ClientLease lease;          // Lease information for the requested
+                                    // transaction.  To ensure prepare requests
+                                    // are linearizable.
+        uint64_t ackId;             // Id of the largest RPC id whose metadata
+                                    // can be garbage-collected.  Used for
+                                    // linearizability.
+        uint32_t participantCount;  // Number of all objects participating TX
+                                    // in whole cluster.
+        uint32_t opCount;
+        // List of all Participants of TX.
+        // List of Ops
+
+        struct ReadOp {
+            OpType type;
+            uint64_t tableId;
+            uint64_t rpcId;
+            uint16_t keyLength;
+            RejectRules rejectRules;
+
+            // In buffer: The actual key for this part
+            // follows immediately after this.
+            ReadOp(uint64_t tableId, uint64_t rpcId, uint16_t keyLength,
+                    RejectRules rejectRules)
+                : type(OpType::READ)
+                , tableId(tableId)
+                , rpcId(rpcId)
+                , keyLength(keyLength)
+                , rejectRules(rejectRules)
+            {
+            }
+        } __attribute__((packed));
+
+        struct RemoveOp {
+            OpType type;
+            uint64_t tableId;
+            uint64_t rpcId;
+            uint16_t keyLength;
+            RejectRules rejectRules;
+
+            // In buffer: The actual key for this part
+            // follows immediately after this.
+            RemoveOp(uint64_t tableId, uint64_t rpcId, uint16_t keyLength,
+                       RejectRules rejectRules)
+                : type(OpType::REMOVE)
+                , tableId(tableId)
+                , rpcId(rpcId)
+                , keyLength(keyLength)
+                , rejectRules(rejectRules)
+            {
+            }
+        } __attribute__((packed));
+
+        struct WriteOp {
+            OpType type;
+            uint64_t tableId;
+            uint64_t rpcId;
+            uint32_t length;        // length of keysAndValue
+            RejectRules rejectRules;
+
+            // In buffer: KeysAndValue follow immediately after this
+            WriteOp(uint64_t tableId, uint64_t rpcId, uint32_t length,
+                        RejectRules rejectRules)
+                : type(OpType::WRITE)
+                , tableId(tableId)
+                , rpcId(rpcId)
+                , length(length)
+                , rejectRules(rejectRules)
+            {
+            }
+        } __attribute__((packed));
+    } __attribute__((packed));
+
+    struct Response {
+        ResponseCommon common;
+        Vote vote;
+    } __attribute__((packed));
+};
+
+struct TxRequestAbort {
+    static const Opcode opcode = Opcode::TX_REQUEST_ABORT;
+    static const ServiceType service = MASTER_SERVICE;
+
+    enum Vote { COMMIT, ABORT, INVALID };
```

Comment:
I think you need documentation for this enum. In particular, what does INVALID mean?